src/main/java/com/example/board/global/dto/PageableCustom.java

```diff
@@ -0,0 +1,23 @@
+package com.example.board.global.dto;
+
+import org.springframework.data.domain.PageImpl;
+
+public record PageableCustom(boolean first,
+                             boolean last,
+                             boolean hasNext,
+                             int totalPages,
+                             long totalElements,
+                             int page,
+                             int size) {
+
+    public static <T> PageableCustom of(PageImpl<T> page) {
```

Comment:
```suggestion
    public static <T> PageableCustom of(Page<T> page) {
```

파라미터로 인터페이스를 사용할 수 있다면 인터페이스를 최대한 사용하시는 것이 좋습니다.
구현체로 사용하게 되면 스프링 라이브러리가 버전 업되면서 구현체가 혹시 모를 변경사항이 생겨 서비스에 문제가 생길 가능성이 높아질 수 있을 것 같네요.